Formula/imposm3.rb

```diff
@@ -0,0 +1,23 @@
+class Imposm3 < Formula
+  desc "Imports OpenStreetMap data into PostgreSQL/PostGIS databases"
+  homepage "https://imposm.org"
+  url "https://github.com/omniscale/imposm3/archive/refs/tags/v0.11.1.tar.gz"
+  sha256 "14045272aa0157dc5fde1cfe885fecc2703f3bf33506603f2922cdf28310ebf0"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+  depends_on "geos"
+  depends_on "leveldb"
+
+  ENV["CGO_LDFLAGS"] = "-L#{HOMEBREW_PREFIX}/opt/geos/lib -L#{HOMEBREW_PREFIX}/opt/leveldb/lib"
+  ENV["CGO_CFLAGS"] = "-I#{HOMEBREW_PREFIX}/opt/geos/include -I#{HOMEBREW_PREFIX}/opt/leveldb/include"
+
+  def install
+    ldflags = "-X github.com/omniscale/imposm3.Version=#{version}"
+    system "go", "build", "-ldflags", ldflags, "-o", "#{bin}/imposm", "cmd/imposm/main.go"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/imposm version").chomp
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Testing the functionality of this tool requires a protobuf file (which is binary). I could download it from _any_ source while testing but this add a crucial dependency to this package. What would be the preferred way to solve this on your side?

Reply:
See the `resource` blocks in other formula for a way to downloade extra files in a formula

Reply:
I think I came up with an even better solution. See last commit